sdk/messaging/azservicebus/receiver.go

```diff
@@ -400,5 +541,11 @@ func createLinkOptions(mode ReceiveMode, entityPath string) []amqp.LinkOption {
 }
 
 func isCancelled(err error) bool {
-	return errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded)
+	if errors.Is(err, context.Canceled) ||
+		errors.Is(err, context.DeadlineExceeded) ||
+		err.Error() == "context canceled" { // go-amqp is returning this when I cancel
```

Comment:
I really want to test this in a more controlled way because it sure seems like I should have.

I've got this issue to really take a look at error classification:
https://github.com/Azure/azure-sdk-for-go/issues/15610